_tut/docs/3x/tail/iterant.md

```diff
@@ -0,0 +1,286 @@
+---
+layout: docs3x
+title: Iterant
+type_api: monix.tail.Iterant
+type_source: monix-tail/shared/src/main/scala/monix/tail/Iterant.scala
+description: |
+    Pull-based, compositional streaming data type that can describe lazy, possibly asynchronous events, a purely functional iterator.
+    
+tut:
+  scala: 2.12.4
+  binaryScala: "2.12"
+  dependencies:
+    - io.monix::monix:version3x
+---
+
+## Introduction
+
+The `Iterant` is a data type that describes lazy, possibly asynchronous
+streaming of elements using a pull-based protocol.
+
+It is similar somewhat in spirit to Scala's own
+`collection.immutable.Stream` and with Java's `Iterable`, except
+that it is more composable and more flexible due to evaluation being
+controlled by an `F[_]` data type that you have to supply
+(like [Task](../eval/task.html), [Coeval](../eval/coeval.html)
```

Comment:
`FS2` has a notion of pure streams that we don't support. Maybe it makes sense to suggest using `Coeval` as the least powerful type that can support these use-cases?